scripts/export-git-env-vars.sh

```diff
@@ -2,25 +2,31 @@
 set -euo pipefail
 set -x
 
+KEEP_MY_TAGS_INTACT=${KEEP_MY_TAGS_INTACT:-1}
 
 # In case of running this script on detached head, script has difficulties in finding out
 # what is the current 
 echo "Exporting Git Variables: "
 
-git fetch
-
 function find_most_recent_numeric_tag() {
-    # We use the --prune flag because we've had problems with buildkite agents getting conflicting results here
-    git fetch --tags --prune --prune-tags --force
+
+    if [[ "$KEEP_MY_TAGS_INTACT" != 1 ]]; then
```

Comment:
Note:
```
       -P, --prune-tags
           Before fetching, remove any local tags that no longer exist on the remote if --prune is enabled. This option should be used more carefully, unlike --prune it will remove any
           local references (local tags) that have been created. This option is a shorthand for providing the explicit tag refspec along with --prune, see the discussion about that in
           its documentation.
```

> unlike --prune it will remove any local references (local tags) that have been created

Good catch